aws/resource_aws_emr_cluster.go

```diff
@@ -245,6 +245,11 @@ func resourceAwsEMRCluster() *schema.Resource {
 				ForceNew: true,
 				Optional: true,
 			},
+			"custom_ami_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
```

Comment:
In order to provide a better UX, we should add a plan-time validation following:

```
Length Constraints: Minimum length of 0. Maximum length of 256.

Pattern: [\u0020-\uD7FF\uE000-\uFFFD\uD800\uDC00-\uDBFF\uDFFF\r\n\t]*
```

(As per: https://docs.aws.amazon.com/emr/latest/APIReference/API_RunJobFlow.html#EMR-RunJobFlow-request-CustomAmiId)

Reply:
Done!